pkg/reporegistry/repository.go

```diff
@@ -40,7 +40,11 @@ func LoadAllRepositories(confPaths []string) (rpmmd.DistrosRepoConfigs, error) {
 				// without a dot to separate major and minor release versions
 				distro, err := distroidparser.DefaultParser.Standardize(distroIDStr)
 				if err != nil {
-					return nil, fmt.Errorf("failed to parse distro ID string: %v", err)
+					log.Printf("failed to parse distro ID string, using it as is: %v", err)
+					// NB: Before the introduction of distro ID standardization, the filename
+					//     was used as the distro ID. This is kept for backward compatibility
+					//     if the filename can't be parsed.
+					distro = distroIDStr
```

Comment:
This is more of a question than a comment:
If I understand this change correctly, it means I can have a distro called `bob`, and a repo file called `bob.json`, and it would still work even if I don't register a parser for it?

Reply:
Yes, exactly.